core/src/main/scala/cats/ErrorControl.scala

```diff
@@ -0,0 +1,38 @@
+package cats
+
+import cats.data.EitherT
+
+trait ErrorControl[F[_], G[_], E] extends Serializable {
+  val monadErrorF: MonadError[F, E]
+  val monadG: Monad[G]
+
+  def controlError[A](fa: F[A])(f: E => G[A]): G[A]
+
```

Comment:
seems like a `def control[A,B](fa: F[A])(f: Either[E, A] => G[B]): G[B]` might be a nice function to have. It is more general than controlError and can also be used to implement trial. I think users might often want this variant personally.

The name might not be great, but I would love to see this and have syntax for it.